pool/cqrs_command/assignment_command.ml

```diff
@@ -210,29 +200,38 @@ end = struct
     then Error Pool_common.Message.(AlreadySignedUpForExperiment)
     else
       let* () =
-        Session.is_fully_booked command.session
-        |> function
-        | true -> Error Pool_common.Message.(SessionFullyBooked)
+        match
+          command.waiting_list.Waiting_list.experiment
+          |> Experiment.registration_disabled_value
+        with
+        | true -> Error Pool_common.Message.(RegistrationDisabled)
         | false -> Ok ()
       in
-      match
-        command.waiting_list.Waiting_list.experiment
-        |> Experiment.registration_disabled_value
-      with
-      | true -> Error Pool_common.Message.(RegistrationDisabled)
-      | false ->
-        let create =
-          let open Waiting_list in
-          Assignment.
-            { contact = command.waiting_list.contact
-            ; session_id = command.session.Session.id
-            }
-        in
-        Ok
-          [ Waiting_list.Deleted command.waiting_list |> Pool_event.waiting_list
-          ; Assignment.Created create |> Pool_event.assignment
-          ; Email.Sent confirmation_email |> Pool_event.email
-          ]
+      let* () =
+        CCList.fold_left
```

Comment:
isn't this possible as `CCList.iter`?

Reply:
I don't think so, the signature of `CCList.iter` is `('a -> unit) -> 'a list -> unit`. No result.

It would be possible to use `CCList.map & CCList.all_ok`

Reply:
Nice, I'd prefer `CCList.map Session.assignment_creatable |> CCList.all_ok`.